src/loader/loader.c

```diff
@@ -119,40 +111,91 @@ static shmem_request_hook_type prev_shmem_request_hook;
 #endif
 static ProcessUtility_hook_type prev_ProcessUtility_hook;
 
-/* This is timescaleDB's versioned-extension's post_parse_analyze_hook */
-static post_parse_analyze_hook_type extension_post_parse_analyze_hook = NULL;
+typedef struct TsExtension
+{
+	/*
+	 * Static data
+	 */
+
+	// Name of the extension (must be part of the so file name)
+	char const *const name;
+	// Name of the schema for table_name.
+	char const *const schema_name;
+	// Name of the table whose existence indicates the extension is loaded.
+	char const *const table_name;
+	// Name of the GUC for disabling loading this extension.
+	char const *const guc_disable_load_name;
+
+	/*
+	 * Run-time state
+	 */
+
+	// Current value of this extension's disable GUC.
+	bool guc_disable_load;
+
+	// Shared object library version loaded; empty if none.
+	char soversion[MAX_VERSION_LEN];
```

Comment:
Please use C-style comments.

Reply:
I've switched to C89-style comments to follow https://www.postgresql.org/docs/15/source-format.html

You'll see it (plus rebase) on next push.

Thanks!